src/cursor_manager.ts

```diff
@@ -361,14 +360,24 @@ export class CursorManager implements Disposable {
                 }], isMultiSelection: ${editor.selections.length > 1}`,
             );
 
-            if (selection.isEmpty) {
-                // exit visual mode when clicking elsewhere
-                if (this.main.modeManager.isVisualMode && kind === TextEditorSelectionChangeKind.Mouse)
-                    await this.client.input("<Esc>");
+            if (this.main.modeManager.isInsertMode) {
+                // Keep the cursor consistent as much as possible to avoid
+                // changes in editor selections when synchronizing the cursor
+                // position of neovim to vscode in insert mode, thus breaking
+                // the cursor-related functions of vscode, such as placeholder
+                // jumps in snippet mode, multi-cursor selection, etc.
+                this.setWantInsertCursorUpdate(editor, false);
                 await this.updateNeovimCursorPosition(editor, selection.active);
             } else {
-                if (kind !== TextEditorSelectionChangeKind.Mouse || !config.disableMouseSelection)
-                    await this.updateNeovimVisualSelection(editor, selection);
+                if (selection.isEmpty) {
```

Comment:
nit: you can make this an `else if` chain of 

```ts
if (this.main.modeManager.isInsertMode) {
   // ...
}  else if (selection.isEmpty) {
  // ...
} else {
  // ...
}

Reply:
Style issue, I rarely (and don't like to) use `elseif` for conditions that are not in the same group, as the logical boundaries are unclear. I generally use `switch` or early return instead.